The PostCommit Python Xlang Gcp Dataflow job is flaky

The PostCommit Python Xlang Gcp Dataflow is failing over 50% of the time.
Please visit https://github.com/apache/beam/actions/workflows/beam_PostCommit_Python_Xlang_Gcp_Dataflow.yml?query=is%3Afailure+branch%3Amaster to see all failed workflow runs.
See also Grafana statistics: http://metrics.beam.apache.org/d/CTYdoxP4z/ga-post-commits-status?orgId=1&viewPanel=8&var-Workflow=PostCommit%20Python%20Xlang%20Gcp%20Dataflow

Likely due to #36654 by checking commits of the first failing run (https://github.com/apache/beam/commits/6b4dc555a245b4f82250677de21890955867314c) and last succeeded run (https://github.com/apache/beam/commits/fe07fe767098663c032f84df74b4b4f7723bf607).

There is a Spanner write job stuck (e.g. job id `2025-12-04_12_26_17-394437020649606185`)

Elapsed time 4 hr 9 min

```
"org.apache.beam.sdk.util.UserCodeException: com.google.cloud.spanner.SpannerException: UNAVAILABLE: com.google.api.gax.rpc.UnavailableException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at org.apache.beam.sdk.util.UserCodeException.wrap(UserCodeException.java:39)
	at org.apache.beam.sdk.io.gcp.spanner.ReadSpannerSchema$DoFnInvoker.invokeProcessElement(Unknown Source)
	at org.apache.beam.fn.harness.FnApiDoFnRunner.processElementForWindowObservingParDo(FnApiDoFnRunner.java:639)
...
Caused by: com.google.cloud.spanner.SpannerException: UNAVAILABLE: com.google.api.gax.rpc.UnavailableException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
	at com.google.cloud.spanner.SpannerExceptionFactory.newSpannerExceptionPreformatted(SpannerExceptionFactory.java:396)
	at com.google.cloud.spanner.SpannerExceptionFactory.newSpannerException(SpannerExceptionFactory.java:223)
	at com.google.cloud.spanner.SpannerExceptionFactory.newSpannerException(SpannerExceptionFactory.java:135)
	at com.google.cloud.spanner.SessionPool$WaiterFuture.pollUninterruptiblyWithTimeout(SessionPool.java:1975)
	at com.google.cloud.spanner.SessionPool$WaiterFuture.get(SessionPool.java:1920)
...
```

--- a/.github/workflows/beam_PostCommit_Python_Xlang_Gcp_Dataflow.yml
+++ b/.github/workflows/beam_PostCommit_Python_Xlang_Gcp_Dataflow.yml
@@ -81,7 +81,9 @@ jobs:
         uses: ./.github/actions/gradle-command-self-hosted-action
         with:
           gradle-command: :sdks:python:test-suites:dataflow:gcpCrossLanguagePostCommit
-          arguments: -PuseWheelDistribution
+          arguments: |
+            -PpythonVersion=3.14 \
+            -PuseWheelDistribution \
       - name: Archive Python Test Results
         uses: actions/upload-artifact@v4
         if: failure()

--- a/.github/workflows/beam_PostCommit_Python_Xlang_IO_Dataflow.yml
+++ b/.github/workflows/beam_PostCommit_Python_Xlang_IO_Dataflow.yml
@@ -81,6 +81,7 @@ jobs:
         with:
           gradle-command: :sdks:python:test-suites:dataflow:ioCrossLanguagePostCommit
           arguments: |
+            -PpythonVersion=3.14 \
             -PuseWheelDistribution \
             -PkafkaBootstrapServer=10.128.0.40:9094,10.128.0.28:9094,10.128.0.165:9094 \
       - name: Archive Python Test Results

